.github/workflows/ci.yml

```diff
@@ -0,0 +1,146 @@
+# From https://github.com/sebastianbergmann/phpunit/blob/20ab19d3aed56fccf9569cd33c6cd0baab0ec272/.github/workflows/ci.yml
+# (With many modifications)
+
+on:
+  pull_request:
+    branches:
+      - main
+      - master 
+  push:
+    branches:
+      - main
+      - master
+
+name: CI
+
+jobs:
+  coding-guidelines:
+    name: Coding Guidelines
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Install PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: 7.3
+          coverage: none
+
+      - name: Install dependencies
+        run: composer install --no-ansi --no-interaction --no-progress
+
+      - name: Run linter
+        run: composer run cs
+
+  tests:
+    name: Tests
+
+    runs-on: ${{ matrix.os }}
+
+    env:
+      PHP_EXTENSIONS: dom, json, libxml, mbstring, pdo_sqlite, soap, xml, xmlwriter
+      SHELL_INTERACTIVE: true
+
+    strategy:
+      matrix:
+        os:
+          - ubuntu-latest
+          - windows-latest
+
+        php-version:
+          - "7.4"
+
+        php-ini-values:
+          - assert.exception=1, zend.assertions=1
+
+        dependencies:
+          - locked
+        
+        codecov:
+          - false
+
+        include:
+          - os: ubuntu-latest
```

Comment:
I remember the problem. In theory, this included scenario should inherit `dependencies: locked`, but I found that I had to define dependencies explicitly in each include block, or it wouldn't get set.